test/messaging/MessageProcessor.t.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.28;
+
+import {IAuth} from "../../src/misc/interfaces/IAuth.sol";
+
+import {MessageProcessor} from "../../src/core/messaging/MessageProcessor.sol";
+import {IScheduleAuth} from "../../src/core/messaging/interfaces/IScheduleAuth.sol";
+import {IMessageProcessor} from "../../src/core/messaging/interfaces/IMessageProcessor.sol";
+
+import "forge-std/Test.sol";
+
+contract TestCommon is Test {
+    address immutable ANY = makeAddr("any");
+    address immutable AUTH = makeAddr("auth");
+
+    MessageProcessor processor;
+    IScheduleAuth immutable scheduleAuth = IScheduleAuth(makeAddr("ScheduleAuth"));
+
+    function setUp() external {
+        processor = new MessageProcessor(scheduleAuth, AUTH);
+    }
+}
+
+contract TestAuthChecks is TestCommon {
+    function testErrNotAuthorized() public {
+        vm.startPrank(ANY);
+
+        bytes memory EMPTY_MESSAGE;
+
+        vm.expectRevert(IAuth.NotAuthorized.selector);
+        processor.handle(1, EMPTY_MESSAGE);
+
+        vm.stopPrank();
+    }
+}
+
+contract TestFile is TestCommon {
+    function testErrNotAuthorized() public {
+        vm.prank(address(ANY));
+        vm.expectRevert(IAuth.NotAuthorized.selector);
+        processor.file("gateway", address(0));
+    }
+
+    function testErrFileUnrecognizedParam() public {
+        vm.prank(address(AUTH));
+        vm.expectRevert(IMessageProcessor.FileUnrecognizedParam.selector);
+        processor.file("unknown", address(0));
```

Comment:
Nit: Suppose you decided against a fuzzed string because of the many exceptions we would have to make?